src/avram/nothing.cr

```diff
@@ -4,3 +4,12 @@
 # :nodoc:
 class Avram::Nothing
 end
+
+# Use this value when you want to ignore updating a column
+# in a SaveOperation, for example.
+# ```
+# # Sets value to x.value or ignores it if x.value is nil
+# SaveThing.create!(
+#   value: x.value || Undefined
+# )
+Undefined = Avram::Nothing.new
```

Comment:
I've been spinning on this for a few weeks.

- This is top level global namespace, so if some parser library declares Undefined, it might get weird. Similar to how [monkeypatches are handled](https://github.com/luckyframework/avram/blob/main/src/ext/db/connection.cr)
- The overload error message problem means I want this to be as terse as possible -- I considered proposing `Av::No` but there's really no reason why any single set of characters should be used here.

I don't have any material objection to this, but I don't know about the collision possibilities.

Reply:
Yup, same. It may be *too* generic to be in the global space. And maybe my dreams of making this shorter is a sunk cost fallacy anyway? `Avram::IGNORE` is still 5 character shorter than `Avram::Nothing.new` so we would still get the benefit of a single instance in memory. It shouldn't really be too much shorter. Or `Avram::SKIP` which is even shorter.

Reply:
Avram::SKIP or IGNORE has the benefit of being more transparent, which I appreciate. I guess I'm considering a specific scenario in my mind when I say that, which is assembling an update piecewise. Maybe there is another scenario where a constant like this might be used which I'm forgetting about.

There'd be no harm in having both SKIP and IGNORE, I suppose, or some other name. 🤷‍♂️ 

I think cleaning up the stack trace is a secondary concern. It would be nice if those were readable in the slightest, but it's okay that they aren't and I don't really think that's on Avram.

Reply:
I'll give `IGNORE` a try. I think you're right that it is a bit more transparent.